thorvg/examples/thorvg-example/main/display_main.c

```diff
@@ -0,0 +1,221 @@
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_shape_set_fill_color failed");
+
+    tvg_res = tvg_canvas_push(canvas, paint);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_canvas_push failed");
+
+    tvg_res = tvg_canvas_draw(canvas);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_canvas_draw failed");
+
+    tvg_res = tvg_canvas_sync(canvas);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_canvas_sync failed");
+
+    argb888_to_rgb565_ppa(ppa_handle, canvas_buf_888, canvas_buf_565);
+    esp_lcd_panel_draw_bitmap(lcd_panel->panel, 0, 0, LOTTIE_SIZE_HOR, LOTTIE_SIZE_VER, canvas_buf_565);
+
+    /* tvg Lottie */
+    animation = tvg_animation_new();
+    ESP_GOTO_ON_FALSE(animation, ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+
+    Tvg_Paint *picture = tvg_animation_get_picture(animation);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+
+    tvg_res = tvg_picture_load(picture, src);
+    ESP_GOTO_ON_FALSE(picture, ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+
+    tvg_res = tvg_picture_set_size(picture, LOTTIE_SIZE_HOR, LOTTIE_SIZE_VER);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+
+    tvg_res = tvg_canvas_push(canvas, picture);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+
+    float f_total;
+    float f = 0;
+    tvg_res = tvg_animation_get_total_frame(animation, &f_total);
+    ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+    ESP_GOTO_ON_FALSE((f_total != 0.0f), ESP_ERR_INVALID_STATE, err, TAG, "tvg_engine_init failed");
+
+    while (f < f_total) {
+        tvg_res = tvg_animation_get_frame(animation, &f);
+        f++;
+        ESP_LOGI(TAG, "set %f / %f", f, f_total);
+        tvg_res = tvg_animation_set_frame(animation, f);
+        ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_animation_set_frame failed");
+
+        tvg_res = tvg_canvas_update(canvas);
+        ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_canvas_update failed");
+
+        tvg_res = tvg_canvas_draw(canvas);
+        ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_canvas_draw failed");
+
+        tvg_res = tvg_canvas_sync(canvas);
+        ESP_GOTO_ON_FALSE(tvg_res == TVG_RESULT_SUCCESS, ESP_ERR_INVALID_STATE, err, TAG, "tvg_canvas_sync failed");
+
+        argb888_to_rgb565_ppa(ppa_handle, canvas_buf_888, canvas_buf_565);
+        esp_lcd_panel_draw_bitmap(lcd_panel->panel, 0, 0, LOTTIE_SIZE_HOR, LOTTIE_SIZE_VER, canvas_buf_565);
+
+        vTaskDelay(pdMS_TO_TICKS(1));
```

Comment:
so the RTOS tick rate will affect the animation frame rate. It's not easy for the user to tune. maybe we can use esp_timer to create a periodic job?

Not a blocker of this PR BTW.

Reply:
> so the RTOS tick rate will affect the animation frame rate. It's not easy for the user to tune. maybe we can use esp_timer to create a periodic job?
> 
> Not a blocker of this PR BTW.

OK, I'll update it.